apps/testingaccessibility/src/components/pricing.tsx

```diff
@@ -83,9 +83,13 @@ export const Pricing: React.FC<PricingProps> = ({
   const fullPrice =
     (formattedPrice?.unitPrice || 0) * (formattedPrice?.quantity || 0)
 
-  const percentOff =
-    appliedMerchantCoupon &&
-    Math.floor(appliedMerchantCoupon.percentageDiscount * 100)
+  const percentOff = appliedMerchantCoupon
+    ? Math.floor(appliedMerchantCoupon.percentageDiscount * 100)
+    : formattedPrice && isDiscount(formattedPrice)
+    ? Math.floor(
+        (formattedPrice.calculatedPrice / formattedPrice.unitPrice) * 100,
+      )
+    : 0
```

Comment:
upgrade discounts are calculated on the fly so if no coupon is present and the calculated process is still a discount it's an upgrade